### PR TITLE
unbound: host overrides: use legacy reconfigure and reduce configd calls

### PR DESCRIPTION
**Pull request by swhite2**
Fixes https://github.com/opnsense/core/issues/5513

Makes sure the 'Apply' button actually does something, since the host overrides are configured via the legacy section.

### CODE REVIEW

File: src/opnsense/service/conf/actions.d/actions_unbound.conf [hunk 1]
```diff
@@ -83,3 +83,10 @@ command:pgrep -q -nF /var/run/unbound.pid && echo "unbound is running" || echo "
 parameters:
 type:script_output
 message:request Unbound status
+
+[reconfigure]
+command:pluginctl -s unbound restart
+parameters:
+type:script
+message:Configuring and restarting Unbound
+description:Configure and restart Unbound via legacy code
```
Comment (fichtner) [inline src/opnsense/service/conf/actions.d/actions_unbound.conf:92]: any reason we need a cron job (description set)?

Comment (swhite2) [reply src/opnsense/service/conf/actions.d/actions_unbound.conf:92]: We don't, misinterpretation on my end :)

File: src/etc/inc/plugins.inc.d/unbound.inc [hunk 2]
```diff
@@ -416,7 +416,7 @@ function unbound_configure_do($verbose = false, $interface = '')
         flush();
     }
 
-    configd_run('unbound start', true);
+    mwexecf('/usr/local/bin/flock -n -E 0 -o /tmp/unbound_start.lock /usr/local/opnsense/scripts/unbound/start.sh');
```
Comment (fichtner) [inline src/etc/inc/plugins.inc.d/unbound.inc:419]:
```suggestion
    mwexec('/usr/local/bin/flock -n -E 0 -o /tmp/unbound_start.lock /usr/local/opnsense/scripts/unbound/start.sh');
```
FWIW ;)

Comment (swhite2) [reply src/etc/inc/plugins.inc.d/unbound.inc:419]: :neutral_face: I need a coffee.

**Comment by swhite2** [comment]
@AdSchellevis The latest commit makes a lot more sense.

File: src/opnsense/service/conf/actions.d/actions_unbound.conf [hunk 3]
```diff
@@ -41,21 +41,19 @@ type:script
...
+command:pluginctl -s unbound stop
 parameters:
 type:script
 message:Stopping Unbound
 
 [restart]
-command:
-  /usr/local/sbin/unbound-control -c /var/unbound/unbound.conf stop &&
-  /usr/local/bin/flock -n -E 0 -o /tmp/unbound_start.lock /usr/local/opnsense/scripts/unbound/start.sh
+command:pluginctl -s unbound stop && pluginctl -s unbound start
```
Comment (fichtner) [inline src/opnsense/service/conf/actions.d/actions_unbound.conf:56]: We want to start even if stop fails

Comment (fichtner) [reply src/opnsense/service/conf/actions.d/actions_unbound.conf:56]: Ps don’t we have “restart” too?

Comment (swhite2) [reply src/opnsense/service/conf/actions.d/actions_unbound.conf:56]: You're right

**Comment by fichtner** [comment]
what I like about this is that unbound_start.lock is now only used from within unbound.inc and only once at that 👍

**Comment by fichtner** [comment]
@swhite2 now after this unbound takes a long time to become available on boot. disabled my blocklist to see if that is part of the cause or not. can you take a closer look too? ideally unbound should be available much sooner.

**Comment by swhite2** [comment]
> @swhite2 now after this unbound takes a long time to become available on boot. disabled my blocklist to see if that is part of the cause or not. can you take a closer look too? ideally unbound should be available much sooner.

Interesting. I'll take a look.

**Pull request merged**